generate_axial_tht.py

```diff
@@ -0,0 +1,756 @@
+                    Vertex(Position(-dx_body, dy_legs), Angle(0)),
+                    Vertex(Position(-dx_body, dy_body), Angle(0)),
+                    Vertex(Position(dx_body, dy_body), Angle(0)),
+                    Vertex(Position(dx_body, dy_legs), Angle(0)),
+                    Vertex(Position(dx_legs, dy_legs), Angle(0)),
+                    Vertex(Position(dx_legs, -dy_legs), Angle(0)),
+                    Vertex(Position(dx_body, -dy_legs), Angle(0)),
+                    Vertex(Position(dx_body, -dy_body), Angle(0)),
+                    Vertex(Position(-dx_body, -dy_body), Angle(0)),
+                    Vertex(Position(-dx_body, -dy_legs), Angle(0)),
+                    Vertex(Position(-dx_legs, -dy_legs), Angle(0)),
+                ]
+
+        # Package outline
+        footprint.add_polygon(Polygon(
+            uuid=_uuid(uuid_ns + 'polygon-outline'),
+            layer=Layer('top_package_outlines'),
+            width=Width(0),
+            fill=Fill(False),
+            grab_area=GrabArea(False),
+            vertices=_create_outline_vertices(),
+        ))
+
+        # Courtyard
+        footprint.add_polygon(Polygon(
+            uuid=_uuid(uuid_ns + 'polygon-courtyard'),
+            layer=Layer('top_courtyard'),
+            width=Width(0),
+            fill=Fill(False),
+            grab_area=GrabArea(False),
+            vertices=_create_outline_vertices(offset=courtyard_excess, around_pads=True),
+        ))
+
+        # Text
+        x = (-variant.pitch / 2) if variant.vertical else 0
+        footprint.add_text(StrokeText(
+            uuid=_uuid(uuid_ns + 'text-name'),
+            layer=Layer('top_names'),
+            height=Height(1.0),
+            stroke_width=StrokeWidth(0.2),
+            letter_spacing=LetterSpacing.AUTO,
+            line_spacing=LineSpacing.AUTO,
+            align=Align('center bottom'),
+            position=Position(x, (body_diameter_nom / 2) + line_width + 0.5),
+            rotation=Rotation(0.0),
+            auto_rotate=AutoRotate(False),
+            mirror=Mirror(False),
+            value=Value('{{NAME}}'),
+        ))
+        footprint.add_text(StrokeText(
+            uuid=_uuid(uuid_ns + 'text-value'),
+            layer=Layer('top_values'),
+            height=Height(1.0),
+            stroke_width=StrokeWidth(0.2),
+            letter_spacing=LetterSpacing.AUTO,
+            line_spacing=LineSpacing.AUTO,
+            align=Align('center top'),
+            position=Position(x, (-body_diameter_nom / 2) - (line_width + 0.5)),
+            rotation=Rotation(0.0),
+            auto_rotate=AutoRotate(False),
+            mirror=Mirror(False),
+            value=Value('{{VALUE}}'),
+        ))
+
+        # 3D model
+        uuid_3d = _uuid('{}{}-3d'.format('v' if variant.vertical else 'h', variant.pitch))
+        name_3d = '{}, {} mm'.format('Vertical' if variant.vertical else 'Horizontal', variant.pitch)
+        if uuid_3d not in generated_3d_uuids:
```

Comment:
Is this your solution to avoid re-generating existing models? With this approach, we could simply put a static version number in the cached UUID identifier, right?

Reply:
No, there are some 3D models within packages which are used by multiple footprints. This condition makes sure that every (unique) 3D model is added only once to the package, not once per footprint. I added a code comment to make this more clear.

Reply:
Ah, makes sense!